schemas/migration.md

```diff
@@ -0,0 +1,165 @@
+IPLD Schemas and Migration
+--------------------------
+
+Fundamental to our approach to schemas is an understanding:
+
+> Data Never Changes.  Only our interpretation varies.
+
+Data can be created under one schema, and interpreted later under another.
+Data may predate or be created without any kind of schema at all.
+All of this needs to be fine.
+
+Moreover, before talking about migration, it's important to note that we
+don't allow the comforting, easy notion that migration is a one-way process,
+or can be carried out atomically at one magically instantaneous point in time.
+Because data is immutable, and producing updated versions of it doesn't make
+the older version of the data go away, migration is less a thing that you do;
+and more a state of mind.  Migration has to be seamless at any time.
+
+Migration comes in two parts:
+
+1. Understand what data we have;
+2. and having a process to map it into the format of data we want.
+
+We'll spend a few sections on part 1, and then get on to part 2.
+
+### Using Schema Match checking as Version Detection
```

Comment:
These headings have inconsistent casing.

My preference (this is not a strong opinion, I mainly want consistency) is for title case for 1st and 2nd level headings where the titles are short enough but this is a long title. The titles could be shortened though, this one could just be "Version Detection" and the next one "Versioning Theory".

Reply:
fwiw, when I first skrived it, "Schema Match" and "Version Detection" got this-is-an-Important-Noun caps... and then "Using" got first-word caps, and so "checking" was just the odd one out.

But I did that without much in the way of supporting context, and leaning on caps that way is not the greatest writing, so I agree with the fixes :thumbs_up: